Source/ASNetworkImageNode.mm

```diff
@@ -684,7 +729,12 @@ - (void)displayDidFinish
      value stored in _currentImageQuality. */
 
     _renderedImageQuality = _currentImageQuality;
-    [self.delegate imageNodeDidFinishDecoding:self];
+ 
+    // Grab the delegate before unlocking the lock
+    __weak id<ASNetworkImageNodeDelegate> delegate = _delegate;
+
+    ASDN::MutexUnlocker u(__instanceLock__);
+    [delegate imageNodeDidFinishDecoding:self];
```

Comment:
Please refactor this so that we don't need the unlocker, because by the end of this scope, we unnecessary lock and then unlock immediately.